src/main/java/org/geonetwork/index/model/record/IndexRecord.java

```diff
@@ -566,7 +569,7 @@ public class IndexRecord {
 
   @JsonIgnore
   @Singular("codelist")
-  private Map<String, ArrayList<Codelist>> codelists = new HashMap<>();
+  final private Map<String, ArrayList<Codelist>> codelists = new HashMap<>();
```

Comment:
However when singular is used the builder is not allowed to do anything, and the field should be marked final.
That is @Builder.Default and @Singular cannot both be used at once...

Reply:
Singular allows to add element one element
Builder provides the default to add the list or map

![Screenshot from 2024-07-24 09-09-17](https://github.com/user-attachments/assets/5f6a0d40-33eb-4b01-8ebe-40970cafacde)

Reply:
Yeah I had to revert this back, not sure how to address the warning caused by providing a new map here.